gql/lib/src/language/lexer.dart

```diff
@@ -162,15 +162,15 @@ class _Scanner {
     }
     consumeWhitespace();
 
-    if (position >= src.length) {
+    final code = peek();
+
+    if (code == null) {
```

Comment:
@klavs this is based on your earlier comment and the change in the initial draft. Though it worth noting as it removes the `position >= src.length` check entirely